cmake/HPX_SetupCUDA.cmake

```diff
@@ -6,57 +6,118 @@
 
 if(HPX_WITH_CUDA AND NOT TARGET Cuda::cuda)
 
-  find_package(CUDA REQUIRED)
+  if(HPX_WITH_CUDA_CLANG AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
```

Comment:
The question now is: do we still need the `HPX_WITH_CUDA_CLANG` option?

Reply:
The compile options were not the same so I assumed so (but this could be replaced with generator expressions). Will tackle this as soon as clang-cuda jenkins CI passes ^^'